core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1599,6 +1607,23 @@ object Parser extends ParserInstances {
       override def parseMut(state: State): B = Impl.map(parser, fn, state)
     }
 
+    final def branch[A, B, C](
+        parser: Parser[Either[A, B]],
+        l: Parser[A => C],
+        r: Parser[B => C],
+        state: State
+    ): C = {
+      val either = parser.parseMut(state)
+      if ((state.error eq null) && state.capture)
+        either.fold(a => l.ap(Parser.pure(a)), b => r.ap(Parser.pure(b))).parseMut(state)
+      else null.asInstanceOf[C]
```

Comment:
I think this isn't right. I think if we are not capturing we still need to run the result because it may advance the offset in the parser.